### PR TITLE
UPSTREAM: 57967: Fixed TearDown of NFS with root squash.

### PR DESCRIPTION
**Pull request by jsafrane**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1518237

/assign @rootfs @gnufied

### CODE REVIEW

**Comment by jsafrane** [comment]
flake: #17901 ?

**Comment by jsafrane** [comment]
/retest
flake: #17901
hmm, it flakes too much. Did I break it?

**Comment by jsafrane** [comment]
/assign @eparis @mfojtik 
for approval

File: vendor/k8s.io/kubernetes/pkg/util/mount/mount.go [hunk 1]
```diff
@@ -208,6 +209,12 @@ func IsNotMountPoint(mounter Interface, file string) (bool, error) {
 	// IsLikelyNotMountPoint provides a quick check
 	// to determine whether file IS A mountpoint
 	notMnt, notMntErr := mounter.IsLikelyNotMountPoint(file)
+	if notMntErr != nil && os.IsPermission(notMntErr) {
```
Comment (mfojtik) [inline vendor/k8s.io/kubernetes/pkg/util/mount/mount.go]: i think you don't need to check for nil, but that is a nit ;-)

**Comment by jsafrane** [comment]
flake: #18317
/retest

**Pull request merged**